addons/sale_project/models/sale_order.py

```diff
@@ -249,9 +249,7 @@ def action_view_project_ids(self):
             return {'type': 'ir.actions.act_window_close'}
 
         sorted_line = self.order_line.sorted('sequence')
-        default_sale_line = next((
-            sol for sol in sorted_line if sol.product_id.type == 'service'
-        ), self.env['sale.order.line'])
+        default_sale_line = next((sol for sol in sorted_line if sol.product_id.detailed_type == 'service'), None)
```

Comment:
But now since your changes are equivalent to the current code, I would just add your test in that version. 🙂 

Sorry if it was not clear